v6.4 Changelog + upgrade notes (#5394)
mattermost/docs: source/upgrade/important-upgrade-notes.rst

+----------------------------------------------------+------------------------------------------------------------------------------------------------------------------------------------------------------------------+
| If you’re upgrading from a version earlier than... | Then...                                                                                                                                                          |
+====================================================+==================================================================================================================================================================+

> | v6.4                                               | A new schema migration system has been introduced. Before updating the app to this version, it is strongly recommended to have a database backup. The new        |

Reviewer: Same question about "app" here -- what "app" is being referred to?

Author: Changed it to "server".

> | v6.4                                               | A new schema migration system has been introduced. Before updating the app to this version, it is strongly recommended to have a database backup. The new        |

Reviewer: ```suggestion
| v6.4                                               | A new schema migration system has been introduced. Before updating the app to this version, we strongly recommend backing up the database. The new        |
```

|                                                    | migration system will run through all existing migrations to record them to a new table. This will only happen for the first run in order to migrate the         |

> |                                                    | application to the new system. The table where the migration information is stored is ``db_migrations``. Additionally, a ``db_lock`` table is used to prevent    |

Reviewer: ```suggestion
|                                                    | application to the new system. The table where the migration information is stored is called ``db_migrations``. Additionally, a ``db_lock`` table is used to prevent    |
```

+----------------------------------------------------+------------------------------------------------------------------------------------------------------------------------------------------------------------------+
| If you’re upgrading from a version earlier than... | Then...                                                                                                                                                          |
+====================================================+==================================================================================================================================================================+

> | v6.4                                               | New schema migrations                                                                                                                                            |

Author: @isacikgoz @agnivade @streamer45 
1. Please help provide texts for the new schema migrations for important upgrade notes once ready.
2. Please also help review the [Administration changes](https://github.com/mattermost/docs/blob/570f3c8455b550d4bf141540bb53f360243c2ab6/source/install/self-managed-changelog.md#administration) section in the changelog draft to see if anything else needs to be added to the important upgrade notes.

Reviewer 2: @isacikgoz - Could we provide Amy with a detailed note? Things that I'd like to see are:
- What to expect?
- How to debug if something went wrong? Talk a bit about db_migrations table here.

Author: I've added Ibrahim's [proposed text](https://github.com/mattermost/docs/pull/5394#pullrequestreview-869543303) (with minor wording tweaks) to the changelog and important upgrade notes.

Author: @agnivade @isacikgoz RC-1 for v6.4 was recently cut. Let me know if additional testing for the new migration system is needed either on community-daily or v6.4-RC1.

Reviewer 2: I'll defer to Ibrahim, but I don't think it's necessary.
